src/pages/ChatPage.tsx

```diff
@@ -0,0 +1,349 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import {
+  Message,
+  MessageEventChannel,
+  MessageService,
+} from '@adorsys-gis/message-service';
+import DeleteIcon from '@mui/icons-material/Delete';
+import MoreVertIcon from '@mui/icons-material/MoreVert';
+import { v4 as uuidv4 } from 'uuid';
+
+const ChatPage: React.FC = () => {
+  const { contactId } = useParams<{ contactId: string }>();
+  const navigate = useNavigate();
+
+  const [contactName, setContactName] = useState<string>('');
+  const [contactDID, setContactDID] = useState<string>('');
+  const [messages, setMessages] = useState<Message[]>([]);
+  const [newMessage, setNewMessage] = useState<string>('');
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [deleteOptionsVisible, setDeleteOptionsVisible] = useState<{
+    [key: string]: boolean;
+  }>({});
+
+  // Memoize service creation
+  const contactService = useMemo(() => new ContactService(eventBus), []);
+  const messageService = useMemo(() => new MessageService(eventBus), []);
+
+  useEffect(() => {
+    const fetchContactDetails = async () => {
+      if (contactId) {
+        const id = parseInt(contactId);
+        contactService.getContact(id);
+
+        const handleContactReceived = (response: ServiceResponse<Contact>) => {
+          if (
+            response.status === ServiceResponseStatus.Success &&
+            response.payload
+          ) {
+            setContactName(response.payload.name);
+            setContactDID(response.payload.did);
+            setErrorMessage(null);
+          } else {
+            setErrorMessage('Failed to fetch contact details.');
+          }
+        };
+
+        const getContactChannel = ContactEventChannel.GetContactByID;
+        eventBus.on(getContactChannel, handleContactReceived);
+
+        return () => {
+          eventBus.off(getContactChannel, handleContactReceived);
+        };
+      } else {
+        setErrorMessage('Contact ID is undefined.');
+      }
+    };
+
+    fetchContactDetails();
+  }, [contactId, contactService]);
+
+  useEffect(() => {
+    const fetchMessages = async () => {
+      if (contactId) {
+        messageService.getAllMessagesByContact(contactDID);
+
+        const handleMessagesReceived = (
+          response: ServiceResponse<Message[]>,
+        ) => {
+          if (
+            response.status === ServiceResponseStatus.Success &&
+            response.payload
+          ) {
+            setMessages(response.payload);
+            setErrorMessage(null);
+          } else {
+            setErrorMessage('Failed to fetch messages.');
+          }
+        };
+
+        const getAllByContactIdChannel = MessageEventChannel.GetAllByContactId;
+        eventBus.on(getAllByContactIdChannel, handleMessagesReceived);
+        return () => {
+          eventBus.off(getAllByContactIdChannel, handleMessagesReceived);
```

Comment:
Please review how messages are fetched and displayed. When a new message is added and the browser is refreshed, the messages seem to get swapped, which is not ideal. Ensure that new messages appear in the same order they were added.